clippy_lints/src/danger_not_accepted.rs

```diff
@@ -0,0 +1,402 @@
+                    // scenarios so why treat them as strings here?
+                    let token::LitKind::Str | token::LitKind::StrRaw(_) |
+                        token::LitKind::CStr | token::LitKind::CStrRaw(_) = lit.kind
+                => (tok, lit),
+                // Otherwise, raise an error.
+                Some(tokenstream::TokenTree::Token(tok, _)) => {
+                    cx.sess().span_err(tok.span, "expected a string literal after a reason attribute");
+                    return (Vec::new(), None);
+                },
+                _ => {
+                    cx.sess().span_err(eq_tok.span, "expected a string literal after a reason attribute");
+                    return (Vec::new(), None);
+                },
+            };
+
+            specified_reason = Some(reason_lit);
+
+            // Finally, because the reason must be specified as the last attribute, expect either an
+            // optional comma or an EOS and break.
+            match stream.next() {
+                Some(tokenstream::TokenTree::Token(tok, _)) if tok.kind == token::TokenKind::Comma => {
+                    // The token after the comma must be an EOS
+                    if stream.next().is_some() {
+                        cx.sess().span_err(tok.span, NOTHING_AFTER_ERR);
+                    }
+                },
+                None => {},
+                // Otherwise, raise an error.
+                Some(tokenstream::TokenTree::Token(tok, _)) => {
+                    cx.sess().span_err(tok.span, NOTHING_AFTER_ERR);
+                    return (Vec::new(), None);
+                },
+                _ => {
+                    cx.sess().span_err(reason_tok.span, NOTHING_AFTER_ERR);
+                    return (Vec::new(), None);
+                },
+            }
+
+            break;
+        }
+    }
+
+    (dangers, specified_reason)
+}
+
+fn emit_dangerous_call_lint(cx: &LateContext<'_>, expr: &'_ Expr<'_>, unaccepted_dangers: &[UnacceptedDanger]) {
+    // Define formatting helpers
+    struct FmtInline<F>(F);
+
+    impl<F> fmt::Display for FmtInline<F>
+    where
+        F: Fn(&mut fmt::Formatter<'_>) -> fmt::Result,
+    {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            self.0(f)
+        }
+    }
+
+    fn fmt_inline<F>(f: F) -> FmtInline<F>
+    where
+        F: Fn(&mut fmt::Formatter<'_>) -> fmt::Result,
+    {
+        FmtInline(f)
+    }
+
+    // Collect all unique dangers
+    let unique_dangers = unaccepted_dangers
+        .iter()
+        .map(|danger| danger.id)
+        .collect::<FxHashSet<_>>();
+
+    // Create a lint
+    span_lint_and_then(
+        cx,
+        DANGER_NOT_ACCEPTED,
+        expr.span,
+        &format!(
+            "called a function marked with `#[clippy::dangerous(...)]` without blessing the calling \
+             module with `#![clippy::accept_danger({})]`",
+            fmt_inline(|f| {
```

Comment:
this could just map the dangers to a `Vec<String>` and then `<[T]>::join` them instead